podcasts/Onboarding/Plus/PlusPurchaseModal.swift

```diff
@@ -26,39 +26,46 @@ struct PlusPurchaseModal: View {
 
         let firstProduct = products.first
         _selectedOption = State(initialValue: selectedPrice == .yearly ? coordinator.plan.yearly : coordinator.plan.monthly)
-        _freeTrialDuration = State(initialValue: firstProduct?.offer?.duration)
+        _selectedOffer = State(initialValue: firstProduct?.offer)
+    }
+
+    private func price(for subscriptionInfo: PlusPricingInfoModel.PlusProductPricingInfo) -> String {
+        var price = subscriptionInfo.price
+        if let offer = subscriptionInfo.offer {
+            let period = subscriptionInfo.identifier.productInfo.frequency.description
+            if offer.type == .freeTrial {
+                price =  L10n.subscriptionFrequencyPricingFormat(subscriptionInfo.rawPrice, period)
+            } else {
+                price = L10n.subscriptionFrequencyPricingFormat(offer.price, period)
+            }
+        }
+        return price
```

Comment:
Instead of using the nested if's what about using a guard + switch?

```suggestion
        guard let offer = subscriptionInfo.offer else {
            return subscriptionInfo.price
        }

        let period = subscriptionInfo.identifier.productInfo.frequency.description

        switch offer.type {
        case .freeTrial:
            return L10n.subscriptionFrequencyPricingFormat(subscriptionInfo.rawPrice, period)
        case .discount:
            return L10n.subscriptionFrequencyPricingFormat(offer.price, period)
        }
```